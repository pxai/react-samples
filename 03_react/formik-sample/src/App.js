import React from 'react';
import logo from './logo.svg';
import './App.css';
import SimpleForm from './SimpleForm';
import ValidationForm from './ValidationForm';

function App() {
  return (
    <div className="App">
      <header className="App-header">
        <SimpleForm />
        <ValidationForm />
      </header>
    </div>
  );
}

export default App;
