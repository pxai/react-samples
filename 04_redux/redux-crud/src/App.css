body {
 padding: 1em;
}
.App {
  text-align: justify;
  margin: 1em;
}

.App-logo {
  animation: App-logo-spin infinite 20s linear;
  height: 40px;
}

.App-header {
  background-color: #222;
  height: 150px;
  padding: 20px;
  color: white;
}

.App-title {
  font-size: 1.5em;
}

.App-intro {
  font-size: large;
}

@keyframes App-logo-spin {
  from { transform: rotate(0deg); }
  to { transform: rotate(360deg); }
}

.Note {
  padding: 1em;
  margin: 0.5em;
  border: 1px solid lightgrey;
  border-radius: 1em;
}

.NoteControls {
  width: 30%;
  float: right;
}
