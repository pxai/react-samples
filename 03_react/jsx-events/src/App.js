import React, { Component } from "react";
import "./App.css";

class App extends Component {
  arrowHandleClick = (e) => {
    e.preventDefault();
    alert("The first link was clicked.");
  }

  handleClick(e) {
    e.preventDefault();
    alert("The second link was clicked.");
  }

  render() {

    return (
      <div >

      <button onClick={() => alert("It works")}>
          Click on this button
      </button><br />

      <a href="#" onClick={this.arrowHandleClick}>
        Click first link
      </a><br />

      <a href="#" onClick={(e) => this.handleClick(e)}>
        Click second link
      </a><br />
      </div>
    );
  }
}

export default App;
