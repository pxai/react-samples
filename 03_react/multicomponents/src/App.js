import React, { Component } from "react";
import logo from "./logo.svg";
import "./App.css";
import Header from "./Header"
import Paragraph from "./Paragraph"

class App extends Component {
    showMsg (msg) {
        alert("This button works: " + msg);
    }
  render() {
    return (
      <div className="App">
        <Header logo={logo} />
        <Paragraph shMsg={this.showMsg} />
      </div>
    );
  }
}

export default App;
