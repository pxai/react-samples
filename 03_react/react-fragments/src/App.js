import React from 'react';
import './App.css';
import Paragraphs from './Paragraphs';

const phrases = ["Code must read like well-written prose",
	"If you need to use comments you should consider refactoring",
	"Test early, and test often"];

class App extends React.Component {
	render () {
		return (
			<div className="App">
				<header className="App-header">
					React Fragments Sample
					<Paragraphs phrases={phrases}/>
				</header>
			</div>
		);
	}
}

export default App;
