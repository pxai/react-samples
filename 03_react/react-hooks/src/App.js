import React from 'react';
import Clock from "./Clock";
import './App.css';
import ClockEffect from './ClockEffect';
import Countdown from './Countdown';

function App () {
	return (
		<div className="App">
			<header className="App-header">
				React Hooks Sample
				<Clock start={0}/>
				<ClockEffect start={42}/>
				<Countdown start={10}/>
			</header>
		</div>
	);
}

export default App;
