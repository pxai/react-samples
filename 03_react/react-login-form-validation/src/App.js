import React, { Component } from "react";
import "./App.css";

class App extends Component {

  constructor(props) {
    super(props);
    this.state = { login: "", password: "",
                   loginValid: true, passwordValid: true }
  }

  handleSubmit=  (e) => {
    e.preventDefault();
    this.setState({
      login: "",
      password: "",
      loginValid: true,
      passwordValid: true
    })
  }

    loginCheck = (event) => {
        const target = event.target;

        this.setState({
            [target.name]: target.value,
            loginValid: this.isLoginValid(target.value)
        });
    }

    passwordCheck = (event) => {
        const target = event.target;

        this.setState({
            [target.name]: target.value,
            passwordValid: this.isPasswordValid(target.value)
        });
    }

    isLoginValid = (login) => (login.length > 2);
    isPasswordValid = (password) => (password.length > 7);


    handleSubmit = (event) => {
        event.preventDefault();
        if (!this.isLoginValid(this.state.login) ||
            !this.isPasswordValid(this.state.password))
          alert("Form fields are incorrect");
        else
          alert("Form fields correct");
    }

  render() {
    console.log(this.state.loginValid);
    let loginError = (this.state.loginValid)?"":<div className="error">Login not valid</div>;
    let passwordError = (this.state.passwordValid)?"":<div className="error">Password not valid</div>;

    return (
      <div className="App">
        <header className="App-header">
          <h1 className="App-title">Login form</h1>
        </header>
        <div>
          <form onSubmit={this.handleSubmit}>
          <div><label>Login</label></div>
              {loginError}
            <div>
            <input type="text" name="login" onBlur={this.loginCheck} />
          </div>

          <div><label>Password</label></div>
              {passwordError}
            <div>
            <input type="password" name="password" onBlur={this.passwordCheck}  />
          </div>
          <div>
            <input type="submit" value="Login" />
          </div>
          </form>
          </div>
          <div>Login: {this.state.login}, password: {this.state.password}
            </div>
      </div>
    );
  }
}

export default App;