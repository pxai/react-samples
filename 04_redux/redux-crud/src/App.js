import React, { Component } from "react"
import { Route, Link, Switch } from "react-router-dom"
import reduxLogo from "./img/redux.svg"
import "./App.css"
import  Notes  from "./components/Notes"
import  NoteForm  from "./components/NoteForm"
import  NotFound  from "./components/NotFound"
import  About  from "./components/About"

class App extends Component {
  render() {
    return (
      <div className="app">
        <div className="app-header">
            <div className="lapp-header-title">
              <h1>React Redux CRUD sample</h1>
              <img src={reduxLogo} className="App-logo" alt="redux logo" />
            </div>
            <div>
              <Link to="/">Home</Link> | 
              <Link to="/notes/new">New Note</Link> | 
              <Link to="/about">About</Link> | 
            </div>
            <Switch>
              <Route exact path="/" component={Notes}/>
              <Route exact path="/notes/new" component={NoteForm}  />
              <Route exact path="/notes/edit/:id" component={NoteForm}  />
              <Route exact path="/404" component={NotFound}/>
              <Route path="/notes/:id" component={Notes} />
              <Route exact path="/about" component={About} />
              <Route component={NotFound} />
          </Switch>
          
        </div>
      </div>
    );
  }
}


export default App