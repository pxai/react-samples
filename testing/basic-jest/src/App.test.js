import React from 'react';
import ReactDOM from 'react-dom';
import App from './App';
import renderer from 'react-test-renderer';

test('renders without crashing', () => {
  const div = document.createElement('div');
  ReactDOM.render(<App />, div);
  ReactDOM.unmountComponentAtNode(div);
});


test('it matches snapshot', () => {
	const component = renderer.create(<App />);
 let tree = component.toJSON();
  expect(tree).toMatchSnapshot();
});
