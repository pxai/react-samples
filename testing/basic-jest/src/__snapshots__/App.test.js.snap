// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`it matches snapshot 1`] = `
<div
  className="App"
>
  <header
    className="App-header"
  >
    <img
      alt="logo"
      className="App-logo"
      src="logo.svg"
    />
    <h1
      className="App-title"
    >
      Welcome to React
    </h1>
  </header>
  <div
    className="Greet"
    onClick={[Function]}
  >
    This is a greet
  </div>
  <p
    className="App-intro"
  >
    To get started, edit 
    <code>
      src/App.js
    </code>
     and save to reload.
  </p>
</div>
`;

exports[`it matches snapshot 2`] = `
<div
  className="App"
>
  <header
    className="App-header"
  >
    <img
      alt="logo"
      className="App-logo"
      src="logo.svg"
    />
    <h1
      className="App-title"
    >
      Welcome to React
    </h1>
  </header>
  <div
    className="Greet"
    onClick={[Function]}
  >
    This is a greet
  </div>
  <p
    className="App-intro"
  >
    To get started, edit 
    <code>
      src/App.js
    </code>
     and save to reload.
  </p>
</div>
`;

exports[`playground 1`] = `
<div
  className="App"
>
  <header
    className="App-header"
  >
    <img
      alt="logo"
      className="App-logo"
      src="logo.svg"
    />
    <h1
      className="App-title"
    >
      Welcome to React
    </h1>
  </header>
  <div
    className="Greet"
    onClick={[Function]}
  >
    This is a greet
  </div>
  <p
    className="App-intro"
  >
    To get started, edit 
    <code>
      src/App.js
    </code>
     and save to reload.
  </p>
</div>
`;

exports[`playgrounf 1`] = `
<div
  className="App"
>
  <header
    className="App-header"
  >
    <img
      alt="logo"
      className="App-logo"
      src="logo.svg"
    />
    <h1
      className="App-title"
    >
      Welcome to React
    </h1>
  </header>
  <div
    className="Greet"
    onClick={[Function]}
  >
    This is a greet
  </div>
  <p
    className="App-intro"
  >
    To get started, edit 
    <code>
      src/App.js
    </code>
     and save to reload.
  </p>
</div>
`;
